test/e2e/scaffold/scaffold.go

```diff
@@ -583,11 +606,38 @@ func (s *Scaffold) CreateVersionedApisixPluginConfig(yml string) error {
 	return s.CreateResourceFromString(ac)
 }
 
-func (s *Scaffold) CreateVersionedApisixRoute(yml string) error {
-	if !strings.Contains(yml, "kind: ApisixRoute") {
-		return errors.New("not a ApisixRoute")
+func (s *Scaffold) CreateVersionedApisixResource(yml string) error {
+	if strings.Contains(yml, "kind: ApisixRoute") {
+		ar := s.replaceApiVersion(yml, s.opts.APISIXRouteVersion)
+		return s.CreateResourceFromString(ar)
+	}
+	if strings.Contains(yml, "kind: ApisixConsumer") {
+		ac := s.replaceApiVersion(yml, s.opts.APISIXConsumerVersion)
+		return s.CreateResourceFromString(ac)
+	}
+	if strings.Contains(yml, "kind: ApisixPluginConfig") {
+		apc := s.replaceApiVersion(yml, s.opts.ApisixPluginConfigVersion)
+		return s.CreateResourceFromString(apc)
 	}
+	return errors.New("this resource does not support")
```

Comment:
would be better to show what kind of resource it is. Use regex to extract the resource kind and return error in default branch of switch/case.

Reply:
+1

Reply:
Updated.